basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java

```diff
@@ -2725,6 +2725,136 @@ public static Long getLongKey(Object key) {
 		return getLong(key);
 	}
 
+	public static URL getUrlOrNull(String urlString) {
+		if ( urlString == null ) return null;
+		try
+		{
+			URL url = new URL(urlString);
+			url.toURI();
+			return url;
+		} catch (Exception exception) {
+			return null;
+		}
+	}
+
+	public static String stripOffQuery(String urlString)
+	{
+		if ( urlString == null ) return null;
```

Comment:
If you have a nice URL object, then just re-create it from the URL 👍 

  new URI(uri.getScheme(),
                   uri.getAuthority(),
                   uri.getPath(),
                   null, // Ignore query
                   uri.getFragment()).toString();

Reply:
I really wanted to find a StringUtils that did this :(

Reply:
Sam - I just want this to be a string match.  I don't want a bunch of conditional code with ports, etc.

Reply:
ok ...... I prefer using the nice URL object because URLs are complex.... but string matching is fine if you know what the URL looks like

Reply:
for example, is the fragment ever used or important ?

Reply:
It looks like org.apache.http.client.utils.URIBuilder does this?

```
    URIBuilder uriBuilder = new URIBuilder(url);
    uriBuilder.removeQuery();

    return uriBuilder.build().toString();
```
 https://stackoverflow.com/a/54032843/3708872

There's some other ideas near the top for just using URI like the top answer which seems like what @ottenhoff suggests https://stackoverflow.com/a/27267375/3708872

Reply:
yeah but i'm understanding chuck's concern from the phone call: the URL can't change at all because it's part of the hash..... so putting the URL through any type of processing doesn't make sense.

Reply:
I took Matt's suggestion, wrapped in a really cranky unit test - and the URIBuilder code passed with flying colors.   It handles al the things I expected it to handle and then some.  I think it tweaked less than a *URL* processor because URIs are more strings than URLs - but they do know how to pitch a query string.  So that turned out nicely.